libs/rest/generated-rest-client/src/main/kotlin/net/corda/restclient/CordaRestClient.kt

```diff
@@ -0,0 +1,90 @@
+import net.corda.restclient.generated.infrastructure.ApiClient
+import okhttp3.OkHttpClient
+import java.security.SecureRandom
+import java.security.cert.X509Certificate
+import javax.net.ssl.SSLContext
+import javax.net.ssl.TrustManager
+import javax.net.ssl.X509TrustManager
+
+
+class CordaRestClient(
+    private val baseUrl: String,
+    private val httpClient: OkHttpClient
+) {
+
+    var certificatesClient = CertificateApi(baseUrl, httpClient)
+    var configurationClient = ConfigurationApi(baseUrl, httpClient)
+    var cpiClient = CPIApi(baseUrl, httpClient)
+    var flowInfoClient = FlowInfoApi(baseUrl, httpClient)
+    var flowManagementClient = FlowManagementApi(baseUrl, httpClient)
+    var helloRestClient = HelloRestApi(baseUrl, httpClient)
+    var hsmClient = HSMApi(baseUrl, httpClient)
+    var keyManagementClient = KeyManagementApi(baseUrl, httpClient)
+    var keyRotationClient = KeyRotationApi(baseUrl, httpClient)
+    var memberLookupClient = MemberLookupApi(baseUrl, httpClient)
+    var memberRegistrationClient = MemberRegistrationApi(baseUrl, httpClient)
+    var mgmAdminClient = MGMAdminApi(baseUrl, httpClient)
+    var mgmClient = MGMApi(baseUrl, httpClient)
+    var networkClient = NetworkApi(baseUrl, httpClient)
+    var rbacPermissionClient = RBACPermissionApi(baseUrl, httpClient)
+    var rbacRoleClient = RBACRoleApi(baseUrl, httpClient)
+    var rbacUserClient = RBACUserApi(baseUrl, httpClient)
+    var virtualNodeClient = VirtualNodeApi(baseUrl, httpClient)
+    var virtualNodeMaintenanceClient = VirtualNodeMaintenanceApi(baseUrl, httpClient)
+
+
+    companion object {
+        const val API_VERSION = "/api/v5_2" // oldest supported version
+        fun createHttpClient(
+            baseUrl: String = "https://localhost:8888",
+            username: String = "admin",
+            password: String = "admin"
+        ): CordaRestClient {
+
+            val urlWithCordaApiVersion = baseUrl + API_VERSION
+
+            val trustAllCerts: Array<TrustManager> = arrayOf(
+                object : X509TrustManager {
+                    override fun checkClientTrusted(chain: Array<X509Certificate>, authType: String) {}
+                    override fun checkServerTrusted(chain: Array<X509Certificate>, authType: String) {}
+                    override fun getAcceptedIssuers(): Array<X509Certificate> {
+                        return arrayOf()
+                    }
+                }
+            )
+
+            val sslContext = SSLContext.getInstance("SSL")
+            sslContext.init(null, trustAllCerts, SecureRandom())
+            val ac = ApiClient.apply {
+                this.username = username
+                this.password = password
+            }
+            val client = ac.builder
+                .sslSocketFactory(sslContext.socketFactory, trustAllCerts[0] as X509TrustManager)
+                .hostnameVerifier { _, _ -> true }
+                .build()
```

Comment:
And btw there's no need to have the variable, bc we modify static fields. What about something like
```suggestion
            ApiClient.apply {
                this.username = username
                this.password = password
            }
            val client = ApiClient.builder
                .sslSocketFactory(sslContext.socketFactory, trustAllCerts[0] as X509TrustManager)
                .hostnameVerifier { _, _ -> true }
                .build()
```